hazelcast/src/main/java/com/hazelcast/core/IMap.java

```diff
@@ -61,6 +64,9 @@
  * <li>Methods, including but not limited to {@code keySet}, {@code values}, {@code entrySet}, return a collection
  * clone of the values. The collection is <b>NOT</b> backed by the map, so changes to the map are <b>NOT</b> reflected
  * in the collection, and vice-versa.</li>
+ * <li>Methods, including but not limited to {@code computeIfPresent}, may behave incorrectly if the value passed to
```

Comment:
Shouldn't this also be in the `computeIfPresent` method's javadoc? Just by glancing on your PR I found out significant difference between `executeOnKeys` and `executeOnKey`, that is very important for us in Jet and we didn't know about it.

Reply:
As far as I can tell, `computeIfPresent` is not provided by Hazelcast, it's provided by the default method on `java.util.concurrent.ConcurrentMap`, so I can't enhance it's javadoc :)

I will read more about `executeOnKeys` and `executeOnKey`.

Reply:
@taburet your comment reminded me that maybe we can clarify more by adding something like this to javaDoc 
`it is based on default implementation and entire method invocation is not performed atomically`

>may behave incorrectly 

Also we can add an example of incorrect behaviour to javaDoc

Reply:
@viliam-durina We can't amend the javadoc since hazelcast is at JDK 6 and does not support neither default methods nor does it have the `BiFunction` class which is one of its argument types.
@ahmetmircik the issue we are trying to document here is not that it isn't atomic (which you are right about) but that the default `ConcurrentMap.computeIfPresent` loops indefinitely if the provided lambda mutates the returned entry (instead of returning a new instance).